packages/ember-toucan-core/src/-private/assert-block-or-argument-exists.ts

```diff
@@ -0,0 +1,39 @@
+import { assert } from '@ember/debug';
+
+export type AssertBlockOrArg = {
+  blockExists: boolean;
+  argName: string;
+  arg?: string;
+  isRequired?: boolean;
+};
+
+/**
+ * @param {object} AssertBlockOrArg
+ *
+ **/
```

Comment:
Super nit, but I don't think we'll need this JSDoc like this here as TypeScript should handle this all for us in our editors.  It **would** be helpful though to include a comment of what this accomplishes though!

```suggestion
/**
 * This function does the thing!
 **/
```